### ISSUE TITLE
Two identical layers swap styles (when layer refresh is enabled)

### ISSUE THREAD

**Issue by rduivenvoorde** [qgis/QGIS#39831]
See also #39830 

Loading a dynamic layer twice in a project, AND set the automatic refresh to 3 seconds in the layer properties. It seems like the layers are swapping styles every 3 seconds.
In this project:  [stylefixissue.zip](https://github.com/qgis/QGIS/files/5500637/stylefixissue.zip) there are 3 layers.
2 of them are of a service (loaded as File!) which returns a random point on earth on every request: https://duif.net/geotest/points.geojson

If you load this layer, you will see the green dot and red start (both in their own layer!) swap places every 3 seconds, OR emerge on the same place like screendumps below: 
![Screenshot-20201106111603-556x273](https://user-images.githubusercontent.com/731673/98354973-ccfbdd80-2021-11eb-85de-b53f060176cb.png)
![Screenshot-20201106111620-559x273](https://user-images.githubusercontent.com/731673/98354979-cec5a100-2021-11eb-875f-a6e1c7d446d8.png)

I had a theory that it maybe had something todo with QGIS (cleveryly) sharing datasources or so and mixing up the layers???

**Comment by rduivenvoorde** [comment]
If I add a label with the (internal) layer_id 

[stylefixissue.zip](https://github.com/qgis/QGIS/files/6050095/stylefixissue.zip)

you can even see the layers swapping position:

![Screenshot-20210226150737-509x320](https://user-images.githubusercontent.com/731673/109310149-73bff000-7844-11eb-9ee3-b03ee0d61b27.png)

![Screenshot-20210226150753-496x328](https://user-images.githubusercontent.com/731673/109310166-79b5d100-7844-11eb-87c6-df80cffde368.png)

I tried to debug the renderer but that one 'uses the right layer with the right color/style'.

Is it possible that the featuresource or iterator is mixing things up?

@wonder-sk you maybe a clue or pointer?

**Comment by gioman** [comment]
> https://duif.net/geotest/points.geojson

@rduivenvoorde this do not seems to be running anymore, is this still valid?

**Comment by rduivenvoorde** [comment]
Yes, I enabled the points.geojson service again and tested: still swapping :-)

**Comment by gioman** [comment]
Thanks!

**Comment by Pedro-Murteira** [comment]
Still valid on QGIS 3.22.3.
